lib/mapObjects/CQuest.h

```diff
@@ -17,49 +17,27 @@ VCMI_LIB_NAMESPACE_BEGIN
 
 class CGCreature;
 
-class DLL_LINKAGE CQuest final
+class DLL_LINKAGE CQuest: public Rewardable::Limiter
```

Comment:
Perhaps use composition instead of inheritance?

1) Don't see any actual *extensions* of Limiter.
2) Quest *has* completion condition. But quest is not a completion condition

Also this might not work as expected in some edge cases - e.g. rewardable loader will always instantiate Rewardable::Limiter as sublimiters